### PR TITLE
Big Bad Bed Upbed

### PR DESCRIPTION
**Pull request by alsoandanswer**
- Better bed, bedsheet and linen bin sprites

- Also added some additional bedsheets , functionally similar, of course. ( Have fun, mappers! )

- Added more functionality to bedsheets. 
Click to roll them and unroll them to tuck yourself in, 
Alt+Click to fold them, 
Drag and Drop onto yourself to pick them up, 
and you can equip it in your backpack slot to wear it as a cape.
(and look like a goober.)

also you can use screwdrivers on it to make a spooky ghost costume (equipped on overwear slot)

ezgif com-optimize

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

**Comment by alsoandanswer** [comment]
!wip

File: code/game/objects/structures/bedsheet_bin.dm [hunk 1]
```diff
@@ -8,24 +8,159 @@ LINEN BINS
...
+/obj/item/weapon/bedsheet/AltClick(mob/user as mob)
+	if(!istype(loc,/mob))
+		toggle_fold(user)
+	else
+		user.show_message("<span class='warning'>Drop \the [src] first.</span>")
+		..()
 	add_fingerprint(user)
+
+/obj/item/weapon/bedsheet/MouseDrop(mob/user as mob)
+	if((user && (!( user.restrained() ) && (!( user.stat ) && (user.contents.Find(src) || in_range(src, user))))))
```
Comment (fernerr) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
	if((user && (!use_check(user))) && (user.contents.Find(src) || in_range(src, user)))
```
Seems to solve a runtime.

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/structures/bedsheet_bin.dm [hunk 2]
```diff
@@ -8,26 +8,179 @@ LINEN BINS
...
+		toggle_fold(user)
+	else
+		user.show_message("<span class='warning'>Drop \the [src] first.</span>")
+		..()
 	add_fingerprint(user)
+
+/obj/item/weapon/bedsheet/MouseDrop(mob/user as mob)
+	if((user && (!use_check(user))) && (user.contents.Find(src) || in_range(src, user)))
+		if(!istype(user, /mob/living/carbon/slime) && !istype(user, /mob/living/simple_animal))
+			if( !usr.get_active_hand() )		//if active hand is empty
```
Comment (skull132) [inline code/game/objects/structures/bedsheet_bin.dm]: `user` instead of `usr`

**Comment by alsoandanswer** [comment]
!review

**Comment by alsoandanswer** [comment]
ah dear what the fuck did i do this time

**Pull request merged**